### PR TITLE
Register

### PR DESCRIPTION
**Pull request by TiagoPR**
Why:

* First-time user needs to register and redirect to weekly task page.

This change addresses the need by:

* Reading the input and the click of the submit button. Doing the post request to create and login an user. Using react-router-dom to redirect the user to the weekly task page after registering. Passing the JWT token through the local storage.

### CODE REVIEW

File: src/App.js
Comment (zamith) [inline src/App.js:6]: Instead, I think it makes more sense to have an `API_URL` environment variable that you can set to `http://localhost:3000`, then in each call, you can use only the path, or set these consts.

File: src/App.js
Comment (DavideSilva) [inline src/App.js:54]: There's a small improvement we could make here cc: @MFJess 

[JS has short-circuit evaluation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND#short-circuit_evaluation), meaning it stops evaluating when it finds a `false` value in an AND comparison

As example, in `if (false && ...)` it doesn't matter what else is after the `&&` when there's already a `false` value

In here, you are first comparing if the email is equal to the confirmation (same thing for password) and only after you are checking if even you had an email value to begin with. If we submitted the email fields empty, we would be doing comparisons with empty values before we realize the email is empty. As alternative, we could first check if the email value exists and only after we make the check that it matches the confirmation

In this particular case, it's not that really relevant but it's something to keep in mind when you start working with objects and accessing properties inside, for example, `if (user && user.name == "davide")`. If you had this order reversed, you'd get an error when the `user` is not defined because you'd be trying to access the `name` property of `null`

File: src/App.js
Comment (davelange) [inline src/App.js:13]: If you're going to make several API requests from different components, it makes it easier to move the common code out of the component. 
It's a common pattern to have POSTs that send a body (like you're doing twice here), so it helps to have a generic `post` and `get` function, where you can just pass in the path and body, something like:
```
const response = await post(`/users`, {
  email: mail,
  password: pass,
})
```
and the `post` function deals with setting the headers, preparing the body, setting the API base URL, etc, so you don't have to do that every time.

One way to do this is with [custom hooks](https://react.dev/learn/reusing-logic-with-custom-hooks). You could have a `useAPI` hook, for example.

File: src/App.js
Comment (davelange) [inline src/App.js:43]: This works, but it gets messy as you add more fields, and have to add ids to everything to get the values. The more correct way of getting form values is using [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData/get). You create a new FormData using the event target that this function receives, and then can access all the values in the form, without the need to set ids.

File: src/App.js
Comment (davelange) [inline src/App.js:92]: It's good practice to add a `name` attribute to form fields, [check here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name). Also, it's required if you want to use FormData

File: .env.sample
Comment (DavideSilva) [inline .env.sample:1]: No reason to have this file be hidden. Also, add the newline at the end to remove this warning from Github

File: src/App.js
Comment (davelange) [inline src/App.js:21]: It's better to use FormData here. The `handleSubmit` function receives an event from the form, and you can get the data from there.
```
const handleSubmit = (event) => {
  event.preventDefault() // <- prevents the page reloading
  const data = new FormData(event.target);
  const mail = data.get("email")
  const pass = data.get("password")
  ...
```
This requires the form inputs to have a `name` attribute instead of id

Comment (TiagoPR) [reply src/App.js:21]: Oh ok, thanks! I misunderstood

**Review by DavideSilva — APPROVED** [review]
Wait for @davelange's approval but it's looking good :shipit:

**Pull request merged**
